ktor-client/ktor-client-plugins/ktor-client-logging/common/src/io/ktor/client/plugins/logging/Logging.kt

```diff
@@ -239,6 +613,34 @@ public val Logging: ClientPlugin<LoggingConfig> = createClientPlugin("Logging",
     ResponseObserver.install(ResponseObserver.prepare { onResponse(observer) }, client)
 }
 
+private fun Url.pathQuery(): String {
+    return buildString {
+        if (encodedPath.isEmpty()) {
+            append("/")
+        } else {
+            append(encodedPath)
+        }
+
+        if (encodedQuery.isEmpty()) return@buildString
+
+        append("?")
+        append(encodedQuery)
+    }
+}
+
+@OptIn(DelicateCoroutinesApi::class)
+private fun calcRequestBodySize(content: Any, headers: Headers): Long {
+    check(content is OutgoingContent)
+
+    return when (content) {
+        is OutgoingContent.ByteArrayContent -> content.bytes().size.toLong()
+        is OutgoingContent.ContentWrapper -> calcRequestBodySize(content.delegate(), content.headers)
+        is OutgoingContent.NoContent -> 0
+        is OutgoingContent.ProtocolUpgrade -> 0
+        else -> error("Unable to calculate the size for type ${content::class.simpleName}")
```

Comment:
Could you tell me if the exception is supposed to be handled or printed?

I would return `null` and handle this case explicitly

Reply:
It's more like a guard. If the exception is thrown then it's a bug.

Reply:
Please add the test to make sure the exception is not lost